### PR TITLE
Feature/kubeshark

### PR DESCRIPTION
**Pull request by zghanem0**
*Issue #, if available:*

*Description of changes:*
- added kubeshark addon
- updated the index
- created kubeshark.md for doc

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by elamaran11 — COMMENTED** [review]
@zghanem0 Firstly thankyou so much for the PR. Very good addon that can be added to blueprints. Before i dive deep in to the PR. Following essential things are missing as part of minimum criteria:
1. Doc `index.md` is not updated with your for getting listed in Addons list
2. `mkdocs.yaml` is not updated with your new addon for doc website
3. E2E blueprints example stack is not updated with your addon. Make sure to update the blueprints stack with your addon for E2E and also run the E2E once locally.
4. Make sure you add new unit tests for your addon. Once done let me know i can review again.

File: docs/addons/index.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ The framework currently supports the following add-ons.
 | [`XrayAdotAddOn`](./xray-adot-addon.md)                                   | Deploys ADOT Collector for Xray to receive traces from your workloads.                                                                 | ✅ | ✅ |
 | [`GmaestroAddOn`](./gmaestro.md) | Adds [gMaestro](https://app.granulate.io/gMaestroSignup) cost optimization solution for EKS cluster.                                                                         |
 | [`EksPodIdentityAgentAddOn`](./eks-pod-identity-agent.md)       | [Setting up the EKS Pod Identity Agent](https://docs.aws.amazon.com/en_ca/eks/latest/userguide/pod-id-agent-setup.html)                  | ✅ | ✅ |
-
+| [`KubesharkAddOn`](./kubeshark.md)       | [Deep visibility and monitoring of all API traffic](https://github.com/kubeshark/kubeshark)                  | ✅ | ✅ |
```
Comment (elamaran11) [inline docs/addons/index.md:82]: Please order this in alphabetical order.

File: mkdocs.yml [hunk 2]
```diff
@@ -88,6 +88,7 @@ nav:
     - Vpc Cni: 'addons/vpc-cni.md'
     - AWS XRay: 'addons/xray.md'
     - AWS XRay ADOT: 'addons/xray-adot-addon.md'
+    - Kubeshark: 'addons/kubeshark.md'
```
Comment (elamaran11) [inline mkdocs.yml:91]: likewise, order it alphabetically.

File: docs/addons/kubeshark.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+```sh
+$ kubectl -n kube-system port-forward svc/kubeshark-front 3000:80
+```
+
+
+Open the [dashboard](http://localhost:3000)
+
+Then you should be able to see view like this
+![dashboard](https://raw.githubusercontent.com/kubeshark/assets/master/png/kubeshark-ui.png)
```
Comment (elamaran11) [inline docs/addons/kubeshark.md:80]: Same feedback as patterns repo. Run a sample workload and show traffic monitoring in KubeShark console with screenshots.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@zghanem0 Great work, need some minor changes in both PRS, before i can test and merge this.

File: docs/addons/kubeshark.md (hunk 3 above)
Comment (zghanem0) [reply docs/addons/kubeshark.md:81]: Done!

File: mkdocs.yml (hunk 2 above)
Comment (zghanem0) [reply mkdocs.yml]: Done!

File: docs/addons/index.md (hunk 1 above)
Comment (zghanem0) [reply docs/addons/index.md]: Done!

File: docs/addons/kubeshark.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+
+## Example
+
+The example below shows a Kubeshark query that identifies the traffic flowing from the pod "nginx" in the "default" namespace to "aws.com" and "coredns". The query is writen by [Kubeshark Filter Language (KFL)](https://docs.kubeshark.co/en/filtering#kfl-syntax-reference) is the language implemented inside kubeshark/worker that enables the user to filter the traffic efficiently and precisely.
```
Comment (elamaran11) [inline docs/addons/kubeshark.md:86]: Question, Did you show how to deploy nginx in this setup? If someone tries, they wont have nginx pod running in default namespace. Lets have complete instructions in example.

**Review by elamaran11 — COMMENTED** [review]
@zghanem0 One more minor doc update required

File: docs/addons/kubeshark.md (hunk 4 above)
Comment (zghanem0) [reply docs/addons/kubeshark.md]: Done!.

File: docs/addons/kubeshark.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+3.) Access kubeshark using the below command.
+```
+kubectl -n kube-system port-forward svc/kubeshark-front 3000:80
+```
+
+
+4.) Run Kubeshark query to identify the traffic flow.
+
+As shown below, the Kubeshark query used to identify the traffic flowing from the pod "nginx" in the "default" namespace to "aws.com" and "coredns". The query is writen by [Kubeshark Filter Language (KFL)](https://docs.kubeshark.co/en/filtering#kfl-syntax-reference) is the language implemented inside kubeshark/worker that enables the user to filter the traffic efficiently and precisely.
```
Comment (elamaran11) [inline docs/addons/kubeshark.md:115]: Show this KFL query as plain text.

File: lib/addons/index.ts [hunk 6]
```diff
@@ -1,66 +0,0 @@
-
-export * from './ack';
```
Comment (elamaran11) [inline lib/addons/index.ts:2]: WHy is this file removed, this will break the whole functioning of addons with blueprints. You have to remove only your line.

**Review by elamaran11 — COMMENTED** [review]
Please fix this issue asap. This is the reason for GH action failure.

File: docs/addons/kubeshark.md (hunk 5 above)
Comment (zghanem0) [reply docs/addons/kubeshark.md:118]: done!, i added the KFL query

File: lib/addons/index.ts (hunk 6 above)
Comment (zghanem0) [reply lib/addons/index.ts]: I reverted back this file deletion

**Comment by elamaran11** [comment]
Fix the GitHub action errors.

**Review by elamaran11 — APPROVED** [review]
LGTM. Thankyou for the contribution.

**Comment by elamaran11** [comment]
@zghanem0 Only concern at this point for merging this PR is, the addon is currently in your private org, so we want to know who will maintain the Kubeshark addon if its in your private repo, whats your path to move the addon to a public org like `aws-samples`.

**Comment by elamaran11** [comment]
@zghanem0 Based on [this](https://github.com/aws-samples/cdk-eks-blueprints-patterns/pull/174#issuecomment-2163549619) comment, merging this. Please followup and update the pattern once the addon is moved to aws-samples.
